src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -7188,19 +7188,17 @@ bool SketchObject::insertBSplineKnot(int GeoId, double param, int multiplicity)
                 QT_TRANSLATE_NOOP("Exceptions", "Knot cannot have zero multiplicity."));
     }
 
-    const Part::Geometry* geo = getGeometry(GeoId);
-
+    const auto* const geo{getGeometry(GeoId)};

```

Comment:
I think this `auto` will only make the code harder to read. I'd let the "reader" have a breather here and just be explicit.

Reply:
I can change it if you like, but personally I prefer this shorter form for non-primitives. If I'm unsure of the type I can use the IDE to inspect it by hovering the variable, see screenshot.
<img width="427" alt="image" src="https://github.com/user-attachments/assets/a8fecaa3-49e6-4824-84c1-b330c77fe9d1" />

Reply:
I think our thresholds are just different here. This may be informed by the fact that I find IDEs to be too clunky. Probably when I figure out how to make my spacemacs work properly as an IDE (at the risk of going into jargon, there are some `lsp` layers but they don't work well with my current set up).

Reply:
I've been where you are, but I used to use vim instead.
Everything is possible with vim and emacs, but it's a lot of configs to get it to preform as well as a modern IDE.

I still use vim for quickly changing some configs in files, especially on remote systems but for larger projects I use vscode.
I recommend you to try it out.

In this specific case I think it is quite obvious what the type is, but if you prefer explicit types I  will change it 🙂